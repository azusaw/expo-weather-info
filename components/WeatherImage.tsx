import React from "react";
import { StyleSheet, ViewStyle } from "react-native";
import { Image } from "expo-image";
import { View, Text } from "@/components/Themed";
import { getWeatherDetailFromCode } from "@/libs/getWeatherDetailFromCode";
import { useScreenSizeContext } from "@/components/ScreenSizeProvider";

const WeatherImage = ({ weatherCode }: { weatherCode: number }) => {
  const { isSmall } = useScreenSizeContext();
  const weatherDetail = getWeatherDetailFromCode(weatherCode);
  return (
    weatherDetail && (
      <View style={styles.container}>
        <Image
          style={isSmall ? styles.imageSmall : styles.image}
          source={weatherDetail.image}
          contentFit="cover"
          transition={1000}
        />
        <Text size={14} weight={300}>
          {weatherDetail.description}
        </Text>
      </View>
    )
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
  },
  image: {
    height: 70,
    width: 80,
  },
  imageSmall: {
    height: 50,
    width: 60,
  },
});

export default WeatherImage;
