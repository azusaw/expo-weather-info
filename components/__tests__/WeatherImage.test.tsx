import { render } from "@testing-library/react-native";
import WeatherImage from "@/components/WeatherImage";

describe("WeatherImage", () => {
  test("it renders WeatherImage successfully", () => {
    const { toJSON } = render(<WeatherImage weatherCode={1} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("it returns null with invalid weather code", () => {
    const { toJSON } = render(<WeatherImage weatherCode={1000} />);
    expect(toJSON()).toBeNull();
  });
});
