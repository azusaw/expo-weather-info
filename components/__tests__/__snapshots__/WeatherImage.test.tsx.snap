// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherImage it renders WeatherImage successfully 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "inherit",
      },
      {
        "alignItems": "center",
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={70}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "uri": "http://openweathermap.org/img/wn/01d@2x.png",
        },
      ]
    }
    style={
      {
        "height": 70,
        "width": 80,
      }
    }
    transition={
      {
        "duration": 1000,
      }
    }
    width={80}
  />
  <Text
    style={
      [
        {
          "color": "#888",
          "fontFamily": "Montserrat_300Light",
          "fontSize": 14,
        },
        undefined,
      ]
    }
  >
    Mainly Sunny
  </Text>
</View>
`;
